app/controllers/albums_controller.rb

```diff
@@ -33,6 +33,18 @@ def destroy
     json_response({success: true, message: "Album destroy successfully.", data: {albums: @album}}, 200)
   end
 
+  def mark_as_submitted
+    @album = Album.find_by(id: params[:album_id])
+    if params[:album_id].present?
+      if @album.delivery_status == "Submitted"
+        json_response({success: false, message: "Album already submitted."})
+      else
+        @album.update_attribute :delivery_status, "Submitted"
+        json_response({success: true, message: "Album successfully submitted."}, 201)
+      end
+    end
+  end
+
```

Comment:
Move this action in album controller. and please comment the route before action name.
No need to find album using `@album = Album.find_by(id: params[:album_id])` there is already method fetch_album
And please optimize the code.